plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -546,6 +550,30 @@ else if (prop.size() == 1) {
         }
     }
 
+    private static ColumnMapping dateColumnMappingUsingLocalDate()
+    {
+        return ColumnMapping.longMapping(
+                DATE,
+                dateReadFunctionUsingLocalDate(),
+                dateWriteFunctionUsingLocalDate());
+    }
+
+    private static LongWriteFunction dateWriteFunctionUsingLocalDate()
+    {
+        return (statement, index, value) -> {
+            verifySupportedDate(value);
+            statement.setObject(index, LocalDate.ofEpochDay(value));
+        };
+    }
+
+    private static void verifySupportedDate(long value)
+    {
+        // Deny unsupported dates eagerly to prevent unexpected results. ClickHouse stores '1970-01-01' when the date is out of supported range.
+        if (value < MIN_SUPPORTED_DATE_EPOCH || value > MAX_SUPPORTED_DATE_EPOCH) {
+            throw new TrinoException(INVALID_ARGUMENTS, format("Epoch must be between %s and %s: %s", MIN_SUPPORTED_DATE_EPOCH, MAX_SUPPORTED_DATE_EPOCH, value));
```

Comment:
An epoch is an "instant in time", so I think it makes more sense to say "Date must be between ..."